services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/IslService.java

```diff
@@ -0,0 +1,251 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.event.service;
+
+import org.openkilda.model.Isl;
+import org.openkilda.model.IslStatus;
+import org.openkilda.model.LinkProps;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.SwitchStatus;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.IslRepository;
+import org.openkilda.persistence.repositories.LinkPropsRepository;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Slf4j
+public class IslService {
+    private TransactionManager transactionManager;
+    private IslRepository islRepository;
+    private LinkPropsRepository linkPropsRepository;
+    private SwitchRepository switchRepository;
+
+    public IslService(TransactionManager transactionManager, RepositoryFactory repositoryFactory) {
+        this.transactionManager = transactionManager;
+        this.islRepository = repositoryFactory.createIslRepository();
+        this.linkPropsRepository = repositoryFactory.createLinkPropsRepository();
+        this.switchRepository = repositoryFactory.createSwitchRepository();
+    }
+
+    /**
+     * Create or update isl.
+     *
+     * @param isl isl.
+     */
+    public void createOrUpdateIsl(Isl isl) {
+        log.info("Create or update ISL {}_{}-{}_{}",
```

Comment:
Not sure if it should be info level, we already have information regarding links in OfeLinkBolt.

Reply:
This is fixed. The debug level has been set.